test/userrepo-test.js

```diff
@@ -0,0 +1,75 @@
+const chai = require('chai');
+const expect = chai.expect;
+
+const UserRepo = require('../src/UserRepo').default;
+const User = require('../src/User').default;
+
+describe('UserRepo', function() {
+  let user1, user2, rawUsers, userRepo;
+  beforeEach(function() {
+    user1 = new User ({
+      id: 3000,
+      name: 'Joshua Danger Sevy',
+      address: "3000 Champions Lane, Mars",
+      email: "google@gmail.com",
+      strideLength: 458,
+      dailyStepGoal: 1,
+      friends: [1]
+    });
+    user2 = new User({
+      id: 1,
+      name: '#D0UG13',
+      address: 'Inter-dimensional Palace',
+      email: "google@gmail.com",
+      strideLength: 4802,
+      dailyStepGoal: 4802,
+      friends: [3000]
+    });
+    users = [user1, user2];
+    userRepo = new UserRepo();
+  });
+
+  it('should be a function', function() {
+    expect(UserRepo).to.be.a('function')
+  });
+
+  it('should be an instance of UserRepo', function() {
+    expect(userRepo).to.be.an.instanceof(UserRepo)
+  });
+
+  it('should be able to store data', function() {
+    userRepo.storeData(users)
+
+    expect(userRepo.data).to.deep.equal(users)
+  });
+
+/* THE EXPECTATION HERE IS AN INSTANCEOF RESULT BECAUSE OF THE getRandomNumber
+METHOD IT UTILIZES. THE USER IT GRABS IS ALMOST ALWAYS A DIFFERENT ONE */
+  it('should be able to find the current user', function() {
+    userRepo.storeData(users)
+
+    expect(userRepo.currentUser).to.equal(null)
+    userRepo.findCurrentUser()
+    expect(userRepo.currentUser).to.be.an.instanceof(User)
+  });
+
+  it('should be able to find a user by their ID', function() {
+    userRepo.storeData(users)
+    userRepo.findUserById(3000)
+    let results = userRepo.findUserById(3000)
+
+    expect(results.name).to.equal('Joshua Danger Sevy')
+  });
```

Comment:
Definitely start dropping some 'sad-path' testing in. What happens if the id isn't a number? 

There's a file called 'old-tests' with the original test files if you want to grab and modify tests from there. If it hasn't made it to your local branch let me know and we'll get that sorted.

Reply:
I've been wracking my brain over this, but I don't actually think this test actually has a sad path as there is no actual user interaction with this method. If anything other than a number is passed in, it's because us as the app's devs input something other than a number. I could put in a conditional that'll check if the id isn't a number and will log a phrase to the console as a result, but I don't believe you can't test if a console.log kicked off either. 🤷🏽‍♂️ I'm open to suggestions if you have something in mind, but I think this will end up having to be reverted to the way it is here.